### PR TITLE
Add submission requirement feature to presentation definition matching

### LINKED ISSUE #2540: Support Submission Requirement feature for PE
As described by https://identity.foundation/presentation-exchange/#submission-requirement-feature

This would require changes for both Presentation Definition generation and Presentation Submission checking

### PR DESCRIPTION
**Pull request by woutslakhorst**
Closes #2540 

It looks like there's some duplicate code, but abstracting it with types and interfaces does more harm than good.
This functionality could then process the result of a `merge(...presentationDefinition)` function

### CODE REVIEW

**Comment by woutslakhorst** [comment]
I forgot creating the `PresentationSubmission` when from_nested is used.

File: vcr/pe/submission_requirement.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+		return submissionRequirement.fromNested(availableGroups)
+	}
+	return submissionRequirement.from(availableGroups)
+}
+
+func (submissionRequirement SubmissionRequirement) from(availableGroups map[string]GroupCandidates) ([]vc.VerifiableCredential, error) {
+	selectedVCs := make([]vc.VerifiableCredential, 0)
+	group := availableGroups[submissionRequirement.From]
+	// different rules for 'all' and 'pick'
+	switch submissionRequirement.Rule {
```
[the remark this answers is not in the document]
Comment (woutslakhorst) [reply vcr/pe/submission_requirement.go:67]: Well, easily. I deduplicated using generics, it now has 5 lines of code less... Is it easier to read?

**Review by gerardsn — CHANGES_REQUESTED** [review]
Reviewed before today's changes
Some minor things

File: vcr/pe/submission_requirement.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+		for _, set := range selectedVCs {
+			if len(set) > 0 {
+				count++
+			}
+		}
+		// check count
+		if submissionRequirement.Count != nil {
+			if count < *submissionRequirement.Count {
+				return nil, fmt.Errorf("submission requirement (%s) has less credentials (%d) than requried (%d)", submissionRequirement.Name, count, *submissionRequirement.Count)
+			}
```
[the remark this answers is not in the document]
Comment (woutslakhorst) [reply vcr/pe/submission_requirement.go]: True, but that would be a concern for advanced submission processing?

**Review by gerardsn — APPROVED** [review]

**Pull request merged**